From ee6e8d77da40667cda501b07cda52a8c2142713f Mon Sep 17 00:00:00 2001
From: Pengxuan Zheng <pzheng@codeaurora.org>
Date: Fri, 27 Oct 2017 17:13:50 -0700
Subject: [PATCH 03/19] Turn off sdclang for cfi sanitizer

This patch turns off sdclang for the cfi sanitizer. This is only a temporary
workaround for the cfi sanitizer issue encountered when building O-MR1 with
SDLLVM. It should be reverted once the cfi sanitizer issue is fixed in SDLLVM.

Change-Id: I714916d04a11d63aa6819567be73386320696121
Signed-off-by: Joe Maples <joe@frap129.org>
---
 core/config_sanitizers.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/config_sanitizers.mk b/core/config_sanitizers.mk
index d570ccda8..6d2ecbbca 100644
--- a/core/config_sanitizers.mk
+++ b/core/config_sanitizers.mk
@@ -260,6 +260,10 @@ ifneq ($(my_sanitize),)
 endif
 
 ifneq ($(filter cfi,$(my_sanitize)),)
+  # FIXME: revert this once the cfi sanitizer issue is fixed
+  my_sdclang := false
+  my_sdclang_2 := false
+
   # __cfi_check needs to be built as Thumb (see the code in linker_cfi.cpp).
   # LLVM is not set up to do this on a function basis, so force Thumb on the
   # entire module.
-- 
2.19.1

